assets/js/src/common/set_from_address_modal.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import ReactStringReplace from 'react-string-replace';
+import MailPoet from 'mailpoet';
+
+const showSetFromAddressModal = async () => {
+  MailPoet.Modal.popup({
+    title: MailPoet.I18n.t('setFromAddressModalTitle'),
+    template: ReactDOMServer.renderToString(
```

Comment:
I am not sure about the consequences of using `ReactDOMServer` here. I have never used it before so I am just asking about it. AFAIK its intended purpose is to render React on the server. I am not sure about the usage in the browser. Do you think it is fine?

Reply:
We already use it like this, explained in https://github.com/mailpoet/mailpoet/pull/2680/files#r400026892.